blog/2022-10-17-closure-objects.md

```diff
@@ -0,0 +1,161 @@
+---
+slug: closure-objects
+title: How to remove Java from JavaScript
+authors: [sgranat]
+image: '/img/blog/closure-objects/js-without-java.png'
+description: Replacing JavaScript classes with closure and simple JS constructions. Write React-like code even without React.
+tags: [JavaScript, class, closure, React, TypeScript]
+---
+
+As a React developer, I have an aversion to JS classes. In React bubble, it feels like a bad practice when I see them somewhere.
+
+However, recently I got a task to rewrite our `@tolgee/core`, which is VanillaJS, and so far, there have been classes everywhere. I couldn't stand it, we shouldn't use classes, that's bad. But then I realized that I don't actually know how to replace classes outside of React. Hmm ...
```

Comment:
So you are basically saying. I made my opinion on wrong arguments. I hated classes, but then I found out I need them, so I started to look for a way how to replace them with something not-classy, to be able to stay with my opinion. This is not a good way how to advocate your solution.

You should precisely describe, why you consider classes bad practice at first. Not just describe how to replace them.

Reply:
You can keep it agressive. But there is actually no explanation why classes are bad in the article except the Crockfords citation and that you don't want to write keywords like Java guys, which is both bullshit. And Crokford is criticised a lot for that. You can use it as a joke, but in the same time you should advocate why you consider these concepts wrong. Why I should stop using Classes when I don't hate Java?

Reply:
Ok

Reply:
I've added an section about this